src/Evolution/Systems/RadiationTransport/M1Grey/TimeDerivative.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/RadiationTransport/M1Grey/Fluxes.hpp"
+#include "Evolution/Systems/RadiationTransport/M1Grey/Sources.hpp"
+#include "Evolution/Systems/RadiationTransport/M1Grey/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace RadiationTransport::M1Grey {
+template <typename... NeutrinoSpecies>
+struct TimeDerivative {
+  struct TildeSUp : db::SimpleTag {
+    using type = tnsr::I<DataVector, 3, Frame::Inertial>;
+  };
+
+  using temporary_tags = tmpl::list<TildeSUp>;
+  using argument_tags = tmpl::list<
+      Tags::TildeE<Frame::Inertial, NeutrinoSpecies>...,
+      Tags::TildeS<Frame::Inertial, NeutrinoSpecies>...,
+      Tags::TildeP<Frame::Inertial, NeutrinoSpecies>..., gr::Tags::Lapse<>,
+      gr::Tags::Shift<3>, gr::Tags::SpatialMetric<3>,
+      gr::Tags::InverseSpatialMetric<3>,
+
+      Tags::M1HydroCouplingNormal<NeutrinoSpecies>...,
+      Tags::M1HydroCouplingSpatial<Frame::Inertial, NeutrinoSpecies>...,
+      ::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<3>,
+                    Frame::Inertial>,
+      ::Tags::deriv<gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+                    tmpl::size_t<3>, Frame::Inertial>,
+      ::Tags::deriv<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+                    tmpl::size_t<3>, Frame::Inertial>,
+      gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector> >;
+
+  static void apply(
+      const gsl::not_null<typename Tags::TildeE<
+          Frame::Inertial, NeutrinoSpecies>::type*>... dt_tilde_e,
+      const gsl::not_null<typename Tags::TildeS<
+          Frame::Inertial, NeutrinoSpecies>::type*>... dt_tilde_s,
+
+      const gsl::not_null<typename ::Tags::Flux<
+          Tags::TildeE<Frame::Inertial, NeutrinoSpecies>, tmpl::size_t<3>,
```

Comment:
It's effectively a `std::integral_constant` but for reasons I have forgotten now, brigand defines its own integral_constant and `size_t` is just the `integral_constant<size_t, NUMBER>` type alias. We need to use `tmpl::size_t` because some generic algorithms end up with `tmpl` types instead of `std` types :(

Reply:
Makes sense, thanks!